src/crypto/shelley/ShelleyWallet.js

```diff
@@ -942,7 +944,7 @@ export default class ShelleyWallet extends Wallet implements WalletInterface {
         request.ledgerNanoCatalystRegistrationTxSignData,
       )
 
-      metadata = await catalystUtils.generateRegistration({
+      auxiliaryData = await catalystUtils.generateRegistration({
```

Comment:
Because Registration is one of the possible metadata that is carried inside the auxiliary data, we have the same implementation on yoroi-frontend 
https://github.com/Emurgo/yoroi-frontend/blob/7021ffb4f37ca28b058fcd5ece99ce8c5741bfe9/packages/yoroi-extension/app/api/ada/lib/cardanoCrypto/catalyst.js
Would you suggest a better approach?

Reply:
my issue is about variable names, 
`registration` vs `metadata` vs `auxiliaryData`
generateRegistration should return something called registration

Reply:
wait, why does `generateRegistrationMetadata` return `RustModule.WalletV4.AuxiliaryData`?